data/ai/lua/ca_recruit_rushers.lua

```diff
@@ -1,11 +1,25 @@
 -- Make the generic_recruit_engine functions work as external CAs
 
+local ca_castle_switch
+if (not ai.aspects.passive_leader) then
```

Comment:
This shouldn't be needed, the castle switch CA does this check already and always returns 0 if set.

Reply:
I put that there because it is much faster than going through the ai __cfg to check whether the castle_switch CA is present.  However, since passive_leader is not usually set, and as this is only called once at the beginning anyway, that doesn't make a difference in practice, so I'll remove that conditional.

As for the rest ... Right, I tried to get this done quickly in between other things, which is never a good idea.  (I did test that detecting the CA works though.)  I'll fix this up tomorrow.